### PR TITLE
option to include user agent info to the tenant metrics

### PR DESCRIPTION
**Pull request by ivansenic**
**What this PR does**:
Adds ability to optionally include the user agent to the tenant request counter metrics. By default it's disabled, controlled by the configuration options.

**Which issue(s) this PR fixes**:
Internal issue.

### CODE REVIEW

File: apis/sgv2-quarkus-common/CONFIGURATION.md [hunk 1]
```diff
@@ -46,13 +46,15 @@
...
+| `stargate.metrics.tenant-request-counter.metric-name`            | `String`             | `http.server.requests.counter`      | Name of the metric.                                                                                                                                          |
+| `stargate.metrics.tenant-request-counter.tenant-tag`             | `String`             | `tenant`                            | The tag key for tenant id.                                                                                                                                   |
+| `stargate.metrics.tenant-request-counter.error-tag`              | `String`             | `error`                             | The tag key for the request error flag (true/false).                                                                                                         |
+| `stargate.metrics.tenant-request-counter.user-agent-tag`         | `String`             | `agent`                             | The tag key for the user agent, if capturing is enabled.                                                                                                     |
```
Comment (jeffreyscarpenter) [inline apis/sgv2-quarkus-common/CONFIGURATION.md:56]: maybe `user-agent` as the default to be the most clear about what this is?

File: apis/sgv2-quarkus-common/src/test/java/io/stargate/sgv2/api/common/metrics/TenantRequestMetricsFilterWithUserAgentTagTest.java [hunk 2]
```diff
@@ -0,0 +1,129 @@
...
+          .build();
+    }
+  }
+
+  @Test
+  public void happyPath() {
+    // call endpoint
+    given()
+        .when()
+        .header("user-agent", "python-requests/2.27.1")
```
Comment (jeffreyscarpenter) [inline apis/sgv2-quarkus-common/src/test/java/io/stargate/sgv2/api/common/metrics/TenantRequestMetricsFilterWithUserAgentTagTest.java:56]: nice! I like using actual observed user agent strings in the tests

**Review by jeffreyscarpenter — APPROVED** [review]

File: apis/sgv2-quarkus-common/CONFIGURATION.md (hunk 1 above)
Comment (ivansenic) [reply apis/sgv2-quarkus-common/CONFIGURATION.md:56]: Sure, I can adapt.

Comment (ivansenic) [reply apis/sgv2-quarkus-common/CONFIGURATION.md:56]: @jeffreyscarpenter so usually tags names are defined as:

* `user_agent` - with underscore
* `userAgent` - camel-case

Would like to follow one of those two, any preference?

Comment (jeffreyscarpenter) [reply apis/sgv2-quarkus-common/CONFIGURATION.md:56]: I guess underscore

**Pull request merged**